MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -65,19 +65,40 @@ public class SimulatedLocationManager: NavigationLocationManager {
      */
     @objc public init(route: Route) {
         super.init()
+        initializeSimulatedLocationManager(for: route)
+    }
+
+    /**
+     Initalizes a new `SimulatedLocationManager` with the given route and distance traveled.
+     
+     When location simulation is needed in special cases such as dead reckoning, the distance
+     traveled on the given route determines where the `SimulatedLocationManager` needs to begin the simulation.
+     
+     - parameter route: The initial route.
+     - parameter distanceTraveled: The distance traveled on the current route.
+     - returns: A `SimulatedLocationManager`
+     */
+    @objc public init(route: Route, distanceTraveled: CLLocationDistance) {
```

Comment:
Would it be more versatile if we would pass a `from:CLLocationCoordinate2D,to:CLLocationCoordinate2D`?
As it stands now, we don't stop simulation after simulating through a tunnel if you would get stuck in the middle, right? Slicing the route could possibly fix that.

Reply:
@frederoni a couple of things.
1. We do stop simulation after simulating through a tunnel https://github.com/mapbox/mapbox-navigation-ios/blob/544c5d1822e7e4f9d42809b2220fa555cae684ba/MapboxCoreNavigation/RouteController.swift#L584
2. We don't need to slice as `SimulatedLocationManager` provides us with robust data based on the `currentDistance` (a.k.a. distanceTraveled). Given the `currentDistance`, the manager can determine our current location on the given `route` - which is used to determine factors such as e_xpectedSegmentTravelTimes, closestCoordinateOnRoute, nextCoordinateOnRoute_ and _time_.

https://github.com/mapbox/mapbox-navigation-ios/blob/544c5d1822e7e4f9d42809b2220fa555cae684ba/MapboxCoreNavigation/SimulatedLocationManager.swift#L137

Reply:
👍 
I overlooked this part:
https://github.com/mapbox/mapbox-navigation-ios/blob/544c5d1822e7e4f9d42809b2220fa555cae684ba/MapboxCoreNavigation/RouteController.swift#L583-L585
and thought we would just continue simulating to the destination.